### PR TITLE
Migrate the if-let/when-let to the if-let*/when-let*

### PR DESCRIPTION
**Pull request by sunlin7**
The `if-let`/`when-let` is marked as obsoleted since emacs-26,
 https://lists.gnu.org/archive/html/emacs-devel/2018-03/msg00360.html.

And Emacs-31 produce a warning for if-let/when-let:
https://github.com/emacsmirror/emacs/commit/8903106bb783c2825233c149b6799960aacdea57

So Spacemacs should migrate the if-let/when-let to if-let*/when-let*. 

Please help review the changes. Thanks.

### CODE REVIEW

File: core/libs/quelpa.el [hunk 1]
```diff
@@ -1911,7 +1911,7 @@ version."
                   (package-delete pkg-desc 'force))))
             all-pkgs))
     ;; Only packages with same version remained. Just pick the first one.
-    (when-let (all-pkgs (alist-get name package-alist))
+    (when-let* (all-pkgs (alist-get name package-alist))
```
Comment (bcc32) [inline core/libs/quelpa.el:1914]: This one is wrong, it used the old single-binding syntax and needs to be updated.

Comment (bcc32) [reply core/libs/quelpa.el:1914]: (The rest of the changes LGTM)

**Review by bcc32 — COMMENTED** [review]
Argh, I didn't realize I had to "submit review" to make the review comments show up.

**Comment by bcc32** [comment]
@smile13241324, sorry about this, there is a small typo in the PR that I tried to alert about.

**Comment by sunlin7** [comment]
Hi @bcc32 Thanks for comments, the changes on `core/libs/quelpa.el` actually force restored by the https://github.com/syl20bnr/spacemacs/commit/c3269f6a3b9ce8ee57f362c2b053b5381f8727e3, so it back to original one.

File: core/libs/quelpa.el (hunk 1 above)
Comment (sunlin7) [reply core/libs/quelpa.el:1914]: Restored by https://github.com/syl20bnr/spacemacs/commit/c3269f6a3b9ce8ee57f362c2b053b5381f8727e3

**Pull request merged**